server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java

```diff
@@ -88,16 +97,18 @@
      */
     public static DataType get(int OID) {
         DataType type = lookup.get(OID);
-        return type == null
-                ? UNSUPPORTED_TYPE
-                : type;
+        return type == null ? UNSUPPORTED_TYPE : type;
     }
 
     public static boolean isArrayType(int OID) {
         DataType type = lookup.get(OID);
         return type != null && type.typeElem != null;
     }
 
+    public static boolean isTextForm(int OID) {
+        return notText.get(OID) == null;
```

Comment:
And check whether the item is in the set
```suggestion
        return !notText.contains(OID);
```